File: BarberApp_Nhom6_Be/app/schemas/work_schedule.py
from pydantic import BaseModel, Field
from datetime import time
from typing import Optional

# 🟢 Schema phụ - thông tin stylist
class StylistBrief(BaseModel):
    id: int
    name: str

    class Config:
        from_attributes = True


# 🟢 Schema tạo mới ca làm
class WorkScheduleCreate(BaseModel):
    stylist_id: int
    weekday: str = Field(..., example="Mon")
    start_time: time
    end_time: time


# 🟡 Schema cập nhật ca làm
class WorkScheduleUpdate(BaseModel):
    weekday: Optional[str] = None
    start_time: Optional[time] = None
    end_time: Optional[time] = None


# 🔵 Schema trả dữ liệu ra client
class WorkScheduleOut(BaseModel):
    id: int
    stylist_id: int
    weekday: str
    start_time: time
    end_time: time
    stylist: Optional[StylistBrief] = None  # ✅ thêm thông tin stylist

    class Config:
        from_attributes = True

File: BarberApp_Nhom6_Fe/barberapp_nhom6_fe/lib/models/work_schedule.dart
class WorkSchedule {
  final int id;
  final int stylistId;
  final String weekday; // Mon, Tue, Wed...
  final String startTime;
  final String endTime;
  final String? stylistName;

  WorkSchedule({
    required this.id,
    required this.stylistId,
    required this.weekday,
    required this.startTime,
    required this.endTime,
    this.stylistName,
  });

  factory WorkSchedule.fromJson(Map<String, dynamic> json) {
    return WorkSchedule(
      id: json['id'] as int,
      stylistId: json['stylist_id'] as int,
      weekday: json['weekday'] ?? '',
      startTime: json['start_time'] ?? '',
      endTime: json['end_time'] ?? '',
      stylistName: json['stylist']?['name'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'stylist_id': stylistId,
    'weekday': weekday,
    'start_time': startTime,
    'end_time': endTime,
  };
}
